### DISCUSSION TITLE
[Solved] Config Issues in configuration.yaml & modbusLAN.yaml while setting up in HA.

### DISCUSSION THREAD

**Discussion by gitgrooves** [StealthChesnut/HA-FoxESS-Modbus#77] [General]
Hello and thank you to StealthChesnut and contributors for this awesome HA integration for the Fox ESS inverter!

Initially, I faced difficulties getting the sensor configuration to work in HA using the main branch [configuration.yaml](https://github.com/StealthChesnut/HA-FoxESS-Modbus/blob/main/custom_components/HA-FoxESS-Modbus/configuration.yaml) file, as it seemed to be missing some parts. However, after realising the code was not correctly formatted, I found a solution by using the sensor configuration from a different branch of the repository [here](https://github.com/StealthChesnut/HA-FoxESS-Modbus/blob/336ed8c456212166fdceda99da305c56daef9a00/configuration.yaml).

Furthermore, for my H1-6.0 inverter, the config in the [modbusLAN.yaml](https://github.com/StealthChesnut/HA-FoxESS-Modbus/blob/main/custom_components/HA-FoxESS-Modbus/modbusLAN.yaml) file in the main branch only partially worked. After many hours of troubleshooting, I noticed that the address range in Will's [video](https://youtu.be/uMPr0V6lTHg?t=281) started with 110XX instead of 310XX. Replacing the configuration with the one from a different branch [here ](https://github.com/StealthChesnut/HA-FoxESS-Modbus/blob/336ed8c456212166fdceda99da305c56daef9a00/modbus.yaml) solved this issue and the sensors in HA started working as intended.

Lastly, not sure how I missed this but with out this [USR W610 Setup Guide](https://github.com/StealthChesnut/HA-FoxESS-Modbus/wiki/USR-W610-Setup-Guide) I would have been dead in the water. Life saver! 

Thank you again for this excellent integration as it has been one of the the missing links in getting my inverter data feeding locally into HA.

**Comment by pawel-matyaszczyk** [comment]
@gitgrooves I am struggling with FoxEss Inverter T4 G3 with USR-DP404. Can you give some hint how to debug this thing directly from USR? Any programs you've been working with (modbus, lan, etc?). I've stuck and try to figure out if software inside this inverter need to be upgraded or this is problem with wrong modbus communication.
